### ISSUE TITLE
Handshake failed for nginx ssl option ssl_ecdh_curve secp384r1

### ISSUE THREAD

**Issue by donkristo** [Waboodoo/HTTP-Shortcuts#98]
I've setup my API with nginx server as a reverse proxy following a tutorial for perfect score on ssllabs.com. Using HTTP-Shortcuts I can't connect to my API getting the error "Handshake failed". I tried to disable my custom ssl options one by one to find the issue. And it is the config option "ssl_ecdh_curve secp384r1".

I don't know if this is specific for your app or a general issue with Android but at least I wanted to inform you.

**Comment by Waboodoo** [comment]
Have you tried sending a request from the mobile browser, i.e., sending a simple GET request?

**Comment by donkristo** [comment]
Yes, when using chrome I get the expected response.
I have a LG G6 with Android 7, btw.

**Comment by CV8R** [comment]
I had the exact same setup and was having the same result - "Handshake Failed".

I was in the middle of testing and found one phone we had worked, the other did not.  The difference:

Working
Samsung Galaxy Note 8 SM-N950F Android 8.0.0

Not Working
Samsung Galaxy S7 SM-G930F Android 7.0.

In researching I found.

1. Other apps were affected, not just HTTP-Shortcuts 
2. Updating the S7 to Android 8 resolved it

See the following link as to why: [https://stackoverflow.com/questions/39133437/sslhandshakeexception-handshake-failed-on-android-n-7-0](url)

Cheers
CV8R
